src/overlays/actors/ovl_En_Heishi1/z_en_heishi1.c

```diff
@@ -21,37 +43,473 @@ const ActorInit En_Heishi1_InitVars = {
     (ActorFunc)EnHeishi1_Update,
     (ActorFunc)EnHeishi1_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/EnHeishi1_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/EnHeishi1_Destroy.s")
+f32 sAnimParamsInit[][8] = {
+    { 1.0f, -10.0f, 3.0f, 0.5f, 1000.0f, 200.0f, 0.3f, 1000.0f },
+    { 3.0f, -3.0f, 6.0f, 0.8f, 2000.0f, 400.0f, 0.5f, 2000.0f },
+    { 1.0f, -10.0f, 3.0f, 0.5f, 1000.0f, 200.0f, 0.3f, 1000.0f },
+    { 3.0f, -3.0f, 6.0f, 0.8f, 2000.0f, 400.0f, 0.5f, 2000.0f },
+};
+
+s16 sBaseHeadTimers[] = { 20, 10, 20, 10, 13, 0 };
+
+Vec3f sRupeePositions[] = {
+    { 0.0f, 0.0f, 90.0f },  { -55.0f, 0.0f, 90.0f }, { -55.0f, 0.0f, 30.0f }, { -55.0f, 0.0f, -30.0f },
+    { 0.0f, 0.0f, -30.0f }, { 55.0f, 0.0f, -30.0f }, { 55.0f, 0.0f, 30.0f },  { 55.0f, 0.0f, 90.0f },
+};
+
+s32 sCamDataIdxs[] = {
+    7, 7, 2, 2, 2, 2, 3, 3, 4, 4, 5, 6, 4, 4, 5, 6,
+};
+
+s16 sWaypoints[] = { 0, 4, 1, 5, 2, 6, 3, 7 };
+
+Vec3f sSearchBallAccel = { 0.0f, 0.0f, 0.0f };
+
+Vec3f sSearchBallMult = { 0.0f, 0.0f, 20.0f };
+
+Vec3f sMatrixScale = { 0.3f, 0.3f, 0.3f };
```

Comment:
These look like they should be inline data where they are copied, instead of actual variables.

Reply:
how do I actually do this? If I make it in function static it doesnt work. and if i assign it when declared it happens in the wrong place

Reply:
The variables don't exist, it should be inline wherever the copy happens.
For example:
```c
Vec3f temp = { 0.0f, 0.0f, 0.0f };
```

Reply:
so use brackets to define the vector in the relevant place? cause it doesnt work up top

Reply:
did it and it works, but idk looks kinda weird. pushed the change let me know if its what you had in mind

Reply:
You don't need an additional block since it's already the start of the `if` block.

Reply:
that makes it look a bit better, thanks